### PR TITLE
[WIP] Documentation improvements

### PR DESCRIPTION
**Pull request by karolisc**
I am working on other documentation improvements.
Due to the fact that issues are small and frequent,
I decided to make `WIP` PR to avoid git history pollution.
Later on, I'll squash it o make it tiny.
Besides, it will allow having early comments.

### CODE REVIEW

File: docs/documentation.md [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
   - [Prerequisites](prerequisites.md) - A more detailed view of the prerequisites needed to fully use the `control-repo`
 
-  - [Noop Mode](noop_mode.md) - An overview on how to enforce `noop mode server side` with this repo
+  - [Noop Mode](noop_mode.md) - An overview on how to enforce `server side` `noop mode` with this `control-repo`
```
Comment (alvagante) [inline docs/documentation.md]: I've seen here and in the merged PR that you use single ticks ` to emphasize words.
As said eaerlier I prefer to use monospaced texts (with 1 or 3 ticks, not really sure about the difference if used inline) only to refer to computer strings (paths, commands, outputs) and eventually bold (1 asterisk) to emphasize words. In this case, the original format is already not "compliant" and in other occasions I'd have used an asterisk.
Given that your patches are already big improvements on the existing text, how would YOU emphasize words?

Comment (alvagante) [reply docs/documentation.md]: Actually I meant 2 asterisks to emphasize in bold words which need emphasis

**Comment by karolisc** [comment]
I do not use backticks to emphasize words. According to markdown **asterisks** are meant for [emphasis](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#emphasis) while `backticks` are used for [inline code](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code-and-syntax-highlighting).

Perhaps I do a bit misuse backticks but the main idea behind all this is to group phrases, terms and read/show them as one unit. I saw your efforts to make a similar effect by capitalizing the first letter/letters. I find this approach failing as it is still not clear where the term starts and ends. Another problem that the phrase/term has the same **"weigh"** as the rest of the text and they are lost in the "wall of text" while often phrase/term is the Key/major helper in the sentence.

BTW. this is a good example of the problem, who does the word "helper" refers to in the last sentence?
is it Key helper, major helper OR `Key` and `major helper`. Other examples: CI pipeline or CI Pipeline or `CI pipeline`. Is it Puppet server or Puppet Server or Puppet server( puppet program and some server/node) or I don't care `puppet server`.
Because backticks create boundaries it clarifies all doubts.

#### The rules for emphasis does not change.
To emphasize "marked/highlighted" phrase/term simply put asterisks(**) outside backticks **` term with emphasis `**, `term without emphasis`.

Do not know when one would need to emphasize **emphasized** word but markdown offers the second level of "strong emphasis",  bold with italic ***` term with super emphasis `*** or **_some super emphasis_**

Another benefit of highlighting phrases/terms it can help to reveal text style problems. If the text looks "too" colorful most likely same phrase/term is repeated twice or more times per sentence and review of the text is needed.


#### Regarding **"bolding"** aka emphasis.

This project moderately uses bolding. You also follow common practice of bolding/emphasising ordinary words/terms like  `File`, `Edit`, `Save` ... in [change_process_integration.md](https://github.com/example42/psick/blob/production/docs/change_process_integration.md) 

But do they really worth emphasis? Do they really has same "weight"/importance as **NOTE**, **Important**, **BEWARE**, **WARNING**, **self contained**, **deprecates**, **always enforced**, does **NOT** override etc.? 
Even though this is common internet practice, personally to me it is some long lasting misunderstanding.

File: docs/README.md [hunk 2]
```diff
@@ -23,36 +23,36 @@ Some references and basic information on `Puppet` are provided later, here we st
...
+  - [Fabric](fabric.md) - A review of `Puppet tasks` available with `Fabric`
 
 
 ### Managing changes:
 
-  - [Git tasks](#label-Git) - An overview on how to use `Git`
+  - [Git tasks](git.md) - An overview on how to use `Git`
 
-  - [Change Process](#label-Puppet+change+process) - A `step-by-step` guide on how to manage changes in `Puppet code`
+  - [Change Process](change_process_integration.md) - A `step-by-step` guide on how to manage changes in `Puppet code`
```
Comment (karolisc) [inline docs/README.md]: Most likely originally it was linked to ```change_process.md``` file but I linked it to more detailed process.  
Any ideas how to incorporate  ```change_process.md``` file?

Comment (alvagante) [reply docs/README.md]: Documentation is incomplete (change_process.md), referring to a specific workflow (change_process_integration.md) or  to be updated (workflow.md)...
please use change_process.md here

File: docs/documentation.md [hunk 3]
```diff
@@ -7,20 +7,21 @@
```
Comment (karolisc) [inline docs/documentation.md]: `README.md` has extra [paragraph](https://github.com/example42/psick/tree/production/docs#puppet-control-repo-documentation) but I do not know the purpose of `documentation.md` so I synced only same paragraphs in both pages.

**Pull request merged**